src/test/java/IntegrationTests/WAStorageClientDownloadIT.java

```diff
@@ -0,0 +1,106 @@
+    
+    /**
+     * Test of validateStorageAccount method, of class WAStorageClient.
+     */
+    @Test
+    public void testValidateStorageAccount() throws Exception {
+        System.out.println("validateStorageAccount");
+        StorageAccountInfo storageAccount = testEnv.sampleStorageAccount;
+        boolean expResult = true;
+        boolean result = WAStorageClient.validateStorageAccount(storageAccount);
+        assertEquals(expResult, result);
+    }
+
+    @Test(timeout=1200000)
+    public void testDownloadfromContainer() {
+        System.out.print("download");
+        CloudBlobContainer container;
+        try{
+            CloudStorageAccount account = new CloudStorageAccount(new StorageCredentialsAccountAndKey(testEnv.azureStorageAccountName, testEnv.azureStorageAccountKey1));
+            CloudBlobClient blobClient = account.createCloudBlobClient();
+            String containerName = "testdownload"+TestEnvironment.GenerateRandomString(15);
+            File directory = new File(containerName);
+            directory.mkdir();
+            container = blobClient.getContainerReference(containerName);
+            container.deleteIfExists();
+            container.createIfNotExists();
+            for(int i=0; i<testEnv.TOTAL_FILES;i++){
+                testEnv.downloadFileList.add(new File(directory, "download"+UUID.randomUUID().toString()+".txt"));
+                FileUtils.writeStringToFile(testEnv.downloadFileList.get(i), TestEnvironment.GenerateRandomString(15));
+                LOGGER.log(Level.INFO, testEnv.downloadFileList.get(i).getAbsolutePath());
+                CloudBlockBlob blob = container.getBlockBlobReference(testEnv.downloadFileList.get(i).getName());
+                blob.uploadText(TestEnvironment.GenerateRandomString(20), "UTF-8", AccessCondition.generateEmptyCondition(), null, null);
+            }                         
+            //try to download the same file     
+            Run mockRun = mock(Run.class);
+            Launcher mockLauncher = mock(Launcher.class);
+            WAStorageClient mockStorageClient = spy(WAStorageClient.class);
+            
+            FilePath workspace= new FilePath(mockLauncher.getChannel(), FilenameUtils.getFullPathNoEndSeparator(testEnv.downloadFileList.get(0).getAbsolutePath()));
+            int totalFiles = mockStorageClient.download(mockRun, mockLauncher, TaskListener.NULL, testEnv.sampleStorageAccount, "*.txt", ",archive.zip", "", false, workspace, containerName);
+            
+            assertEquals(testEnv.downloadFileList.size(),totalFiles);
+            container.deleteIfExists();
+            
+        }catch(Exception e){
+            LOGGER.log(Level.SEVERE, null, e);
+            Assert.assertTrue(e.getMessage(), false);
+        }
+    }
+    
+    @After
+    public void tearDown() {
+        System.gc();
+        for (int k = 0; k < testEnv.downloadFileList.size(); k++) {
```

Comment:
if you have a tearDown method I would expect a setUp method as well (the temporary file creation should be there